### PR TITLE
Squash analyzer migrations

### PR DESCRIPTION
**Pull request by ericholscher**
This removes pgvector imports,
allowing us to think about running analyzer tests again.

Currently this is skipping the "heavy" analyzer tests, but re-enabling the basic analyzer tests. 

Depends on https://github.com/readthedocs/ethical-ad-server/pull/857 - can't be merged until that migration is run

### CODE REVIEW

**Comment by ericholscher** [comment]
These tests are now properly running, and showing appropriate failures.

File: adserver/analyzer/backends/__init__.py [hunk 1]
```diff
@@ -1,4 +1 @@
 """Backends for analyzing URLs for keywords and topics."""
-from .eatopics import EthicalAdsTopicsBackend  # noqa
-from .naive import NaiveKeywordAnalyzerBackend  # noqa
-from .textacynlp import TextacyAnalyzerBackend  # noqa
```
Comment (ericholscher) [inline adserver/analyzer/backends/__init__.py:4]: I didn't know why we were doing imports here, but it causes a bunch of issues if we don't have this stuff installed. Seems like a nicer UX to just use their direct paths in the settings, and not force importing them if they aren't used.

Should also save us some memory & startup time on the web servers, which don't need to run this code.

File: config/settings/base.py [hunk 2]
```diff
@@ -520,7 +520,7 @@
 # By default this is empty, you can find options in ``adserver.analyzer.backends``
 ADSERVER_ANALYZER_BACKEND = env.list(
     "ADSERVER_ANALYZER_BACKEND",
-    default=[],
+    default=["adserver.analyzer.backends.naive.NaiveKeywordAnalyzerBackend"],
```
Comment (ericholscher) [inline config/settings/base.py:523]: Not 100% sure we want this, but the requirements for the naive implementation are pretty small, basically just beautifulsoup

**Comment by ericholscher** [comment]
Fixed up the tests, so now they're passing 🎉

File: adserver/analyzer/backends/__init__.py (hunk 1 above)
Comment (davidfischer) [reply adserver/analyzer/backends/__init__.py:4]: Ok. That seems fine.

File: requirements/testing.txt [hunk 3]
```diff
@@ -1,2 +1,3 @@
 # Tox will setup its own environment and so it doesn't depend on anything
 tox<4.0
+beautifulsoup4==4.11.1
```
Comment (davidfischer) [inline requirements/testing.txt:3]: Why is this needed? Shouldn't tox install everything it needs?

**Review by davidfischer — COMMENTED** [review]
I've never [squashed migrations](https://docs.djangoproject.com/en/4.2/topics/migrations/#squashing-migrations) before. Since the previous migration files are gone (from the working copy, not from git), they will all have to be run before the squash migration, correct? So again we have to be very precise in rolling this out.

File: requirements/testing.txt (hunk 3 above)
Comment (ericholscher) [reply requirements/testing.txt:3]: It's currently in the analyzer requirements. Probably makes sense to just move it to the base requirements?

**Comment by ericholscher** [comment]
This is now unblocked.

**Comment by ericholscher** [comment]
@davidfischer I might plan to merge this and deploy it this week if you 👍 it

File: requirements/testing.txt (hunk 3 above)
Comment (davidfischer) [reply requirements/testing.txt:3]: I get it. We need this for the naive analyzer (for tests) but because we aren't installing analyzer requirements this doesn't get installed. Maybe move this one to base but I do understand the desire to have requirements per-analyzer.

File: config/settings/base.py (hunk 2 above)
Comment (davidfischer) [reply config/settings/base.py:523]: By putting it in the default, the analyzer is going to be defaulted to on even for people who take our OSS code. I don't think we want that.. I say we keep the `default=[]` but put this as the default in `testing.py` so the naive analyzer gets tested. Thoughts?

**Review by davidfischer — APPROVED** [review]

File: adserver/analyzer/backends/__init__.py (hunk 1 above)
Comment (ericholscher) [reply adserver/analyzer/backends/__init__.py:4]: These were updated in the prod env 👍

File: requirements/testing.txt (hunk 3 above)
Comment (ericholscher) [reply requirements/testing.txt:3]: Yea.. I'll leave it here for now, but will likely end up in some kind of analyzer base requirements or similar.
